### ISSUE TITLE
[MRAR-17] No RAR packaging (Causes Maven-cobertura-plugin to fail)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-rar-plugin#49]
**[Peter Liljenberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pliljenberg@gmail.com)** opened **[MRAR-17](https://issues.apache.org/jira/browse/MRAR-17?redirect=false)** and commented

For a project with packaging "rar" the codehaus maven plugin for Cobertura does not work.
During instrumentation the following message is displayed:

"Not executing cobertura:instrument as the project is not a Java classpath-capable package"

The reason for this is that in the CoberturaInstrumentMojo.execute()  the code checks which language that the artifact is implemented in, like this:
ArtifactHandler artifactHandler = project.getArtifact().getArtifactHandler();
if ( !"java".equals( artifactHandler.getLanguage() ) )
{
getLog().info( "Not executing cobertura:instrument as the project is not a Java classpath-capable package" );
}

Looking at the components.xml in the Maven sources, we find that the "rar" packaging is not specified at all, meaning that it will be handled with the DefaultArtifactHandler and all properties set to null, including the language property.
This can be fixed with the following addition to components.xml:

\<components>
....
\<component>
\<role>org.apache.maven.artifact.handler.ArtifactHandler\</role>
\<role-hint>rar\</role-hint>
\<implementation>org.apache.maven.artifact.handler.DefaultArtifactHandler\</implementation>
\<configuration>
\<type>rar\</type>
\<extension>rar\</extension>
\<includesDependencies>true\</includesDependencies>
\<language>java\</language>
\<addedToClasspath>false\</addedToClasspath>
\</configuration>
\</component>
...
\</components>

---

**Affects:** 2.3

**Issue Links:**
- [MNG-3147](https://issues.apache.org/jira/browse/MNG-3147) No component for RAR packaging projects in components.xml
 (_**"is duplicated by"**_)
- [MNG-3497](https://issues.apache.org/jira/browse/MNG-3497) rar, par and ejb3 archives should not be added to classpath
 (_**"is superceded by"**_)

1 votes, 3 watchers

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

Wouldn't this artifact definition be more appropriate in the maven-rar-plugin? That way, rar functionality could be captured in a single release cycle, rather than waiting for new releases of Maven itself.

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

We should consider defining an ArtifactHandler and lifecycle mapping for rar packaging...

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

rar lifecycle mapping has been done for MNG-150 in r232602 in august 2005, see http://svn.apache.org/viewvc?view=rev&revision=232602

rar ArtifactHandler definition in components.xml has been done in r593857 on 11/11/2007, released in Maven 2.0.8, see http://svn.apache.org/viewvc?view=rev&revision=593857

but the definition is somewhat different  from your proposal:

```xml
<component>
  <role>org.apache.maven.artifact.handler.ArtifactHandler</role>
  <role-hint>rar</role-hint>
  <implementation>org.apache.maven.artifact.handler.DefaultArtifactHandler</implementation>
  <configuration>
    <type>rar</type>
    <language>java</language>
    <addedToClasspath>true</addedToClasspath>
  </configuration>
</component>
```

instead of

```xml
<component>
  <role>org.apache.maven.artifact.handler.ArtifactHandler</role>
  <role-hint>rar</role-hint>
  <implementation>org.apache.maven.artifact.handler.DefaultArtifactHandler</implementation>
  <configuration>
    <type>rar</type>
    <extension>rar</extension>
    <includesDependencies>true</includesDependencies>
    <language>java</language>
    <addedToClasspath>false</addedToClasspath>
  </configuration>
</component>
```

Then my question is: should the actual configuration be modified?
1. add \<extension>rar\</extension>
2. add \<includesDependencies>true\</includesDependencies>
3. change \<addedToClasspath>true\</addedToClasspath> to \<addedToClasspath>false\</addedToClasspath>
   I'm not a rar expert, so my opinion on the topic is weak :)

BTW, I'll change MNG-3147 to mark it as fixed in Maven 2.0.8
And when we're ok with the changes to do, I'll create another MNG issue

For the dispatch of the diverse ArtifactHandler configurations from Maven Core to corresponding plugins, I think it is another topic. Don't know if it is already covered in an existing Jira issue...

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

rar packaging has been added in Maven 2.0.8
and MNG-3497 is here to fix the config
